packages/react-components/react-avatar/src/components/Avatar/Avatar.types.ts

```diff
@@ -143,15 +77,79 @@ export type AvatarSizes = 20 | 24 | 28 | 32 | 36 | 40 | 48 | 56 | 64 | 72 | 96 |
 /**
  * Properties for Avatar
  */
-export type AvatarProps = Omit<ComponentProps<AvatarSlots>, 'color'> & Partial<AvatarCommons>;
+export type AvatarProps = Omit<ComponentProps<AvatarSlots>, 'color'> & {
+  /**
+   * Optional activity indicator
+   * * active: the avatar will be decorated according to activeAppearance
+   * * inactive: the avatar will be reduced in size and partially transparent
+   * * unset: normal display
+   *
+   * @defaultvalue unset
+   */
+  active?: 'active' | 'inactive' | 'unset';
+
+  /**
+   * The appearance when `active="active"`
+   *
+   * @defaultvalue ring
+   */
+  activeAppearance?: 'ring' | 'shadow' | 'ring-shadow';
+
+  /**
+   * The color when displaying either an icon or initials.
+   * * neutral (default): gray
+   * * brand: color from the brand palette
+   * * colorful: picks a color from a set of pre-defined colors, based on a hash of the name (or idForColor if provided)
+   * * [AvatarNamedColor]: a specific color from the theme
+   *
+   * @defaultvalue neutral
+   */
+  color?: 'neutral' | 'brand' | 'colorful' | AvatarNamedColor;
+
+  /**
+   * Specify a string to be used instead of the name, to determine which color to use when color="colorful".
+   * Use this when a name is not available, but there is another unique identifier that can be used instead.
+   */
+  idForColor?: string | undefined;
+
+  /**
+   * The name of the person or entity represented by this Avatar. This should always be provided if it is available.
+   *
+   * The name will be used to determine the initials displayed when there is no icon, as well as provided to
+   * accessibility tools.
+   */
+  name?: string;
+
+  /**
+   * The avatar can have a circular or square shape.
+   * @defaultvalue circular
+   */
+  shape?: 'circular' | 'square';
+
+  /**
+   * Size of the avatar in pixels.
+   *
+   * Size is restricted to a limited set of supported values recommended for most uses (see `AvatarSizeValue`) and
+   * based on design guidelines for the Avatar control.
+   *
+   * If a non-supported size is neeeded, set `size` to the next-smaller supported size, and set `width` and `height`
+   * to override the rendered size.
+   *
+   * For example, to set the avatar to 45px in size:
+   * `<Avatar size={40} style={{ width: '45px', height: '45px' }} />`
+   *
+   * @defaultvalue 32
+   */
+  size?: AvatarSizes;
+};
 
 /**
  * State used in rendering Avatar
  */
 export type AvatarState = ComponentState<AvatarSlots> &
-  AvatarCommons & {
+  Required<Pick<AvatarProps, 'active' | 'activeAppearance' | 'shape' | 'size'>> & {
     /**
-     * The Avatar's color, with `'colorful'` resolved to a named color
+     * The Avatar's color, it matches props.color but with `'colorful'` resolved to a named color
      */
-    color: Exclude<AvatarCommons['color'], 'colorful'>;
+    color: 'neutral' | 'brand' | AvatarNamedColor;
```

Comment:
Why are we making this change here?

Reply:
Since there's no AvatarCommons, I think she's just copying what `Exclude<AvatarCommons['color'], 'colorful'` represents.

Reply:
Yeah, but we could do the same just using `AvatarProps` instead. which I think would still be better

```suggestion
    color: Exclude<AvatarProps['color'], 'colorful'>;
```